File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):
    answer = []
    
    for i in range(len(arr1)):
        tmp = []
        for j in range(len(arr2[0])):
            tmp_result = 0
            for k in range(len(arr1[0])):
                tmp_result += arr1[i][k] * arr2[k][j]
            tmp.append(tmp_result)
        answer.append(tmp)
        
    return answer

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.js
function solution(arr1, arr2) {
    const answer = [];
    
    for (let i = 0; i < arr1.length; i++) {
        const partialArr = [];
        for (let j = 0; j < arr2[0].length; j++) {
            let result = 0;
            for (let k = 0; k < arr2.length; k++) {
                result += arr1[i][k] * arr2[k][j]
            }
            partialArr.push(result);
        }
        answer.push(partialArr)
    }
    
    return answer;
}


// 2 3 2   5 4 3
// 4 2 4   2 4 1
// 3 1 4   3 1 1

